ipyleaflet/leaflet.py

```diff
@@ -211,8 +219,19 @@ class Velocity(Layer):
     _view_name = Unicode('LeafletVelocityView').tag(sync=True)
     _model_name = Unicode('LeafletVelocityModel').tag(sync=True)
 
+    u_var = Unicode('')
+    v_var = Unicode('')
+    lat_dim = Unicode('latitude')
+    lon_dim = Unicode('longitude')
+    units = Unicode(None, allow_none=True)
+
+    dataset = Instance(Dataset)
     data = Unicode().tag(sync=True)
 
+    def __init__(self, **kwargs):
+        super(Velocity, self).__init__(**kwargs)
+        self.set_trait('data', ds2json(self.dataset, self.u_var, self.v_var, self.lat_dim, self.lon_dim))
```

Comment:
Another way to do this is to not have

```
data = Instance(Dataset).tag(sync=True, to_json=ds_x_to_json, from_json=ds_x_from_json)
```

where `ds_x_to_json` essentially wraps `ds2json`.

Reply:
You would mostly need to remove the `json.dumps` part of the function, which could become private.

Reply:
Great, I didn't know. I'm updating it.

Reply:
Also, there is a warning because traitlets is trying to compare datasets to detect changes.

Reply:
Yeah, the trick would be to add `DataSet` to https://github.com/jupyter-widgets/traittypes

Reply:
(Where we also overrode the set to not use `==`).

Reply:
Traittypes would become a dependency to ipyleaflet. It is already a dependency of pythreejs, ipyvolume and bqplot...

Reply:
And I was not able to observe `dataset` to update `data`, probably because of this comparison issue.